pkg/util/watchdog/watchdog.go

```diff
@@ -48,13 +48,13 @@ func (w *SimpleWatchdog) Start(ctx context.Context) {
 				w.OnError(err)
 			}
 		}
-		if w.hasTickedChan != nil && !channels.IsClosed(w.hasTickedChan) { // On the first tick we close the channel
-			close(w.hasTickedChan)
-		}
 		select {
 		case <-ctx.Done():
 		case <-ticker.C:
 		}
+		if w.hasTickedChan != nil && !channels.IsClosed(w.hasTickedChan) { // On the first tick we close the channel
```

Comment:
I'm pretty sure the idea behind `HasTicked` is to check that `w.onTick` was called once, not that the timer is actually ticked once, see the comment https://github.com/kumahq/kuma/pull/11590/files#r1818950478